runtime/integration-tests/src/generic/cases/lp/mod.rs

```diff
@@ -0,0 +1,1071 @@
+	)
+	.unwrap();
+	env.call_mut(
+		Keyring::Alice,
+		Default::default(),
+		"root",
+		"deny",
+		Some(&[Token::Address(Keyring::Alice.into())]),
+	)
+	.unwrap();
+	env.call_mut(
+		Keyring::Alice,
+		Default::default(),
+		"investment_manager",
+		"deny",
+		Some(&[Token::Address(Keyring::Alice.into())]),
+	)
+	.unwrap();
+	env.call_mut(
+		Keyring::Alice,
+		Default::default(),
+		"pool_manager",
+		"deny",
+		Some(&[Token::Address(Keyring::Alice.into())]),
+	)
+	.unwrap();
+	env.call_mut(
+		Keyring::Alice,
+		Default::default(),
+		"escrow",
+		"deny",
+		Some(&[Token::Address(Keyring::Alice.into())]),
+	)
+	.unwrap();
+	env.call_mut(
+		Keyring::Alice,
+		Default::default(),
+		"user_escrow",
+		"deny",
+		Some(&[Token::Address(Keyring::Alice.into())]),
+	)
+	.unwrap();
+	env.call_mut(
+		Keyring::Alice,
+		Default::default(),
+		"gateway",
+		"deny",
+		Some(&[Token::Address(Keyring::Alice.into())]),
+	)
+	.unwrap();
+	env.call_mut(
+		Keyring::Alice,
+		Default::default(),
+		"pause_admin",
+		"deny",
+		Some(&[Token::Address(Keyring::Alice.into())]),
+	)
+	.unwrap();
+	env.call_mut(
+		Keyring::Alice,
+		Default::default(),
+		"delay_admin",
+		"deny",
+		Some(&[Token::Address(Keyring::Alice.into())]),
+	)
+	.unwrap();
+
+	// ------------------ Substrate Side ----------------------- //
+	// Create router
+	let (base_fee, _) =
+		env.parachain_state(<T as pallet_evm::Config>::FeeCalculator::min_gas_price);
+
+	let evm_domain = EVMDomain {
+		target_contract_address: sp_core::H160::from(env.deployed("router").address().0),
+		target_contract_hash: BlakeTwo256::hash_of(&env.deployed("router").deployed_bytecode),
+		fee_values: FeeValues {
+			value: sp_core::U256::zero(),
+			gas_limit: sp_core::U256::from(500_000),
+			gas_price: sp_core::U256::from(base_fee),
+		},
+	};
+
+	let axelar_evm_router = AxelarEVMRouter::<T>::new(
+		EVMRouter::new(evm_domain),
+		BoundedVec::<u8, ConstU32<MAX_AXELAR_EVM_CHAIN_SIZE>>::try_from(
+			EVM_DOMAIN.as_bytes().to_vec(),
+		)
+		.unwrap(),
+		sp_core::H160::from(env.deployed("router").address().0),
+	);
+
+	env.parachain_state_mut(|| {
+		assert_ok!(
+			pallet_liquidity_pools_gateway::Pallet::<T>::set_domain_router(
+				RawOrigin::Root.into(),
+				Domain::EVM(EVM_DOMAIN_CHAIN_ID),
+				DomainRouter::<T>::AxelarEVM(axelar_evm_router),
+			)
+		);
+	});
+
+	additional(&mut env);
+
+	env.pass(Blocks::ByNumber(1));
+	env
+}
+
+pub fn setup_investment_currencies<T: Runtime>(env: &mut impl EvmEnv<T>) {
+	// Pool 1, Tranche 1
+	// AllowInvestmentCurrency 1
+	// AllowInvestmentCurrency 2
+	// AllowInvestmentCurrency 3
+
+	// Pool 2, Tranche 2
+	// AllowInvestmentCurrency 1
+	// AllowInvestmentCurrency 2
+	// AllowInvestmentCurrency 3
+}
+
+pub fn setup_deploy_lps<T: Runtime>(env: &mut impl EvmEnv<T>) {
+	// ------------------ EVM Side ----------------------- //
+	// Deploy LP and more for both pools and all currencies
+}
+
+pub fn setup_tranches<T: Runtime>(env: &mut impl EvmEnv<T>) {
```

Comment:
```suggestion
pub fn setup_tranches<T: Runtime>(env: &mut impl EvmEnv<T>) {
	unimplemented!("")
```

Reply:
The setups are stackable. Depending on what you want to be set up. This allows to easily test situations. FOr the operations mod I wanted to test that setting up the LP worked at each step that involces messages to be transferred. 

But isolated.

Reply:
Yeah I wanted to point out to you/myself, that the implementation is still missing.

Reply:
https://github.com/moonbeam-foundation/frontier/blob/c5d6daa9ffd46c0a85915526aa26d200fd635e30/frame/ethereum/src/lib.rs#L678-L679